authnz/src/main/java/io/stargate/auth/Resource.java

```diff
@@ -0,0 +1,12 @@
+package io.stargate.auth;
+
+public enum Resource {
+  KEYSPACE,
+  TABLE,
+  TYPE,
+  TRIGGER,
+  FUNCTION,
+  AGGREGATE,
+  VIEW,
+  INDEX
```

Comment:
This looks very similar to our existing `io.stargate.auth.entity.ResourceKind`... Could it be enhanced and reused?

Reply:
Also `AuthorizedResource` might be relevant too.